packages/workflow/src/records/handler/create.ts

```diff
@@ -222,149 +240,385 @@ async function createRecord(
     }
-  } else if (hasValidateScope(request)) {
-    record = await toValidated(record, token)
-    await auditEvent('sent-for-approval', record, token)
-  } else if (hasRegisterScope(request) && !isInProgress(record)) {
-    record = await toWaitingForExternalValidationState(record, token)
-    await auditEvent('waiting-external-validation', record, token)
-  }
-  const eventAction = getEventAction(record)
-  await indexBundle(record, token)
 
-  // Notification not implemented for marriage yet
-  const notificationDisabled =
-    eventAction === 'waiting-external-validation' ||
-    !(await isNotificationEnabled(eventAction, event, token))
+    await indexBundleWithTransaction(readyForReviewRecord, token, transactionId)
 
-  if (!notificationDisabled) {
-    await sendNotification(eventAction, record, token)
+    return
   }
 
   /*
-   * We need to initiate registration for a
-   * record in waiting validation state
+   * You might think, how could validatedRecord be in anything else than READY_FOR_REVIEW state?
+   * If the transaction failed earlier after this point in code
    */
-  if (isWaitingExternalValidation(record)) {
-    const rejectedOrWaitingValidationRecord = await initiateRegistration(
-      record,
-      request.headers,
-      token
+  const validatedRecord = isReadyForReview(readyForReviewRecord)
+    ? await toValidated(readyForReviewRecord, token)
+    : readyForReviewRecord
+
+  await createUserAuditEvent('VALIDATED', {
+    transactionId: transactionId,
+    compositionId: getComposition(validatedRecord).id,
+    trackingId: getTrackingIdFromRecord(validatedRecord),
+    headers: getAuthorizationHeaderFromToken(token)
+  })
+
+  await writeMetricsEvent('sent-for-approval', {
+    record: validatedRecord,
+    authToken: token,
+    transactionId: transactionId
+  })
+
+  await indexBundleWithTransaction(validatedRecord, token, transactionId)
+
+  const action = getEventAction(validatedRecord)
+  const notificationDisabled = !(await isNotificationEnabled(
+    action,
+    event,
+    token
+  ))
+
+  if (!notificationDisabled) {
+    await sendNotification(action, declaredRecord, token)
+  }
+}
+
+export async function registerRecordHandler(
+  recordDetails: BirthRegistration | DeathRegistration | MarriageRegistration,
+  event: EVENT_TYPE,
+  token: PlainToken
+) {
+  const transactionId = `register_${recordDetails.registration?.draftId}`
+  const recordInputWithUploadedAttachments =
+    await uploadBase64AttachmentsToDocumentsStore(
+      recordDetails,
+      getAuthorizationHeaderFromToken(token)
     )
 
-    if (isRejected(rejectedOrWaitingValidationRecord)) {
-      await indexBundle(rejectedOrWaitingValidationRecord, token)
-      await auditEvent('sent-for-updates', record, token)
+  const draftId = recordDetails.registration?.draftId
+
+  const existingComposition = draftId
+    ? await findExistingComposition(draftId, token, [
+        'VALIDATED',
+        'DECLARED',
+        'WAITING_VALIDATION'
```

Comment:
Yea, exactly. One place that got me confused was [this](https://github.com/opencrvs/opencrvs-core/blob/ocrvs-6208/packages/workflow/src/records/handler/create.ts#L150). From my understanding, `IN_PROGRESS` can never happen as the record cannot be incomplete. But it still leaves us with at least `DECLARED`. `VALIDATED` I can't remember where I got from anymore. 

What do you think? Should we keep `DECLARED`? It might be that the `await createRecord` call below goes through but then one of the proceeding calls fail